plugins/GSdx/GS.cpp

```diff
@@ -723,9 +723,13 @@ EXPORT_C_(uint32) GSmakeSnapshot(char* path)
 	{
 		std::string s{path};
 
-		if(!s.empty() && s[s.length() - 1] != DIRECTORY_SEPARATOR)
+		if (!s.empty())
 		{
-			s = s + DIRECTORY_SEPARATOR;
+			// Allows for providing a complete path
+			if (s.substr(s.size() - 4, 4) == ".png")
```

Comment:
Oh shoot. Just noticed this. The basic std::string operator== doesn't compare the contents of the each string. It returns whether the addresses of the two strings are the same.
It would be best to go with .compare(".png") == 0. Furthermore, I'd recommend something like I did here to ensure that it accounts for lowercase & uppercase: https://github.com/PCSX2/pcsx2/blob/2170b6c3a41f00f539123ca982f1b7f9325d1c7d/plugins/GSdx/GSCapture.cpp#L427

Reply:
Hmm, I don't think that's true in C++, see the documentation http://www.cplusplus.com/reference/string/string/operators/ otherwise the normal `Screenshot` option wouldn't work. `==` internally uses `compare`.  But I'll change it to lower-case because why not.  https://repl.it/repls/AnnualTediousGenerics#main.cpp